src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -1050,6 +1057,24 @@ private void EraseNodes()
 
     #region Helpers
 
+    /// <summary>
+    /// Creates a Terminal logger or Console logger based on the environment.
+    /// This method is called by reflection from dotnet. Do not modify the name or parameters without adapting the SDK.
+    /// </summary>
+    public static ILogger CreateTerminalOrConsoleLogger(LoggerVerbosity verbosity)
+    {
+        bool isDisabled = (Environment.GetEnvironmentVariable("MSBUILDTERMINALLOGGER") ?? string.Empty).Equals("off", StringComparison.InvariantCultureIgnoreCase);
+        (bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode) = NativeMethodsShared.QueryIsScreenAndTryEnableAnsiColorCodes();
```

Comment:
nit: we can skip this if TL is requested to be disabled

Reply:
I started with that intention, but this approach has lower complexity and the performance impact is negligible.